### PR TITLE
Crops#search empty resultset (branched off of #1048)

### PR DESCRIPTION
**Pull request by RickCarlino**
# What's New?

 * If an API user gets an empty result set *and* they have passed a `:page` param, the `Crops#index` endpoint will return _all_ crops instead of no crops.
 * If the user did not provide a `:page` param, return `[]` just like before.

# Question Marks

CC: @Br3nda

I think some legacy systems / third party systems may be using this endpoint still and might still expect empty search results to return `[]`. This is one of the oldest endpoints of the app and to be quite honest, I have forgotten a lot of stuff about it since I wrote it almost 5 years ago.

Feedback is appreciated.

### CODE REVIEW

File: spec/controllers/api/api_crops_controller_spec.rb [hunk 1]
```diff
@@ -1,65 +1,75 @@
...
+    @beans = FactoryBot.create(:crop, name: 'mung bean')
     FactoryBot.create_list(:crop, 2)
     Crop.searchkick_index.refresh
   end
 
-  it "lists crops." do
-    skip "this test does not pass on CI - RickCarlino"
-    Legacy._get self, "index", format: :json, filter: "mung"
+  it 'lists crops.' do
+    skip 'this test does not pass on CI - RickCarlino'
```
Comment (Br3nda) [inline spec/controllers/api/api_crops_controller_spec.rb]: Why doesn't it pass?

Comment (Br3nda) [reply spec/controllers/api/api_crops_controller_spec.rb]: How about listing it as pending?

**Comment by Br3nda** [comment]
I plan to test this, on the weekend. Also to see why that test doesn't pass on ci

**Comment by Br3nda** [comment]
47 fixes issues. Nice!

**Comment by RickCarlino** [comment]
@Br3nda Good stuff! I will leave this unmerged for now then (it is up on prod if you want to poke around a live instance, though).

**Comment by Br3nda** [comment]
@RickCarlino this doesn't seem to be deployed. I'm getting an empty recordset here:
https://openfarm.cc/api/v1/crops.json

**Comment by Br3nda** [comment]
ah. I see you need that for backward compatibility.

How about we go to v2 of the api where we don't have that empty record set.

**Comment by RickCarlino** [comment]
Hey @Br3nda, it was a holiday weekend here in the US, so sorry for the latency on replies.

We _could_ go to a V2 of the API, but I realistically don't have enough time in my schedule to support a V2 API on top of a legacy API and the other issues that are outstanding. With the limited time resources we have right now, I think it might be better for us to focus on getting everything upgraded to Rails v6, fixing the ancient Javascript stack and maybe scraping ElasticSearch in favor of MongoDB's full text search (which was not a Mongo feature when the project started).

Working my way through the other issues noted over the weekend now... :eyes:

**Pull request merged**